evap/results/exporters.py

```diff
@@ -122,46 +134,49 @@ def filter_evaluations(semesters, evaluation_states, degrees, course_types, cont
                 continue
             if not evaluation.can_publish_rating_results and not include_not_enough_voters:
                 continue
-            results = OrderedDict()
+            results: OrderedDict[EvaluationResult, list[RatingResult]] = OrderedDict()
             for contribution_result in get_results(evaluation).contribution_results:
                 for questionnaire_result in contribution_result.questionnaire_results:
                     # RatingQuestion.counts is a tuple of integers or None, if this tuple is all zero, we want to exclude it
+                    question_results: list[RatingResult] = questionnaire_result.question_results  # type: ignore[assignment]
```

Comment:
Why is the narrowing to `list[RatingResult]` correct here instead of just `list[QuestionResult]`?

Reply:
We do get RatingResult by convention, and the Typechecker does not like the usage of a more broad Class.

Reply:
> We do get RatingResult by convention
I don't see why that is true. Other result types were not excluded in this method up to this point. The variable is called `question_result`. The very next statement (line 143) goes through all question results and checks whether they are coming from a rating question (-> they are a `RatingResult`) or not. In the end, all results are returned, possibly including other result types.

Where do you see the convention that these are all `RatingResult`s?